### PR TITLE
[FIX] website_sale, website_sale_delivery: update freeshipping on del…

### PR DESCRIPTION
**Pull request by kebeclibre**
…ivery change

Since those two modules do not have a bridge, there was no support for changing the freeshipping line
when changing the delivery carrier

Now it has, at the price of ugly code

OPW 1871526

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kebeclibre** [comment]
goes with https://github.com/odoo/enterprise/pull/2643

File: addons/website_sale/controllers/main.py [hunk 1]
```diff
@@ -1064,3 +1064,7 @@ def country_infos(self, country, mode, **kw):
             states=[(st.id, st.name, st.code) for st in country.get_website_sale_states(mode=mode)],
             phone_code=country.phone_code
         )
+
+    @http.route(['/shop/update_carrier'], type='json', auth='public', methods=['POST'], website=True, csrf=False)
+    def update_eshop_carrier(self, **post):
+        pass
```
Comment (nim-odoo) [inline addons/website_sale/controllers/main.py]: This should become the only `/shop/update_carrier` route. You could have something like:
```
res1 = self._update_website_sale_delivery(...)
res2 = self._update_website_sale_coupon(...)
merge and return results
```

This way, you explicitly control which method is called when, and you don't rely on the installation order.

**Comment by JKE-be** [comment]
je crois que je prefere encore le hack degeulasse de:
    se baser sur la string en version stable que tout ce diff, si t-if online pas possible
    de updater le prix en JS de website_sale_coupon pour website_sale_coupon_delivery

Le diff devrait etre nettement plus court et plus comprehensible

File: addons/website_sale_delivery/controllers/main.py [hunk 2]
```diff
@@ -60,16 +60,27 @@ def update_eshop_carrier(self, **post):
...
+            result = {
+                'status': order.delivery_rating_success,
+                'error_message': order.delivery_message,
+                'carrier_id': carrier_id,
+                'new_amount_delivery': self._format_amount(order.delivery_price, currency),
+                'new_amount_untaxed': self._format_amount(order.amount_untaxed, currency),
+                'new_amount_tax': self._format_amount(order.amount_tax, currency),
+                'new_amount_total': self._format_amount(order.amount_total, currency),
             }
+        response = super(WebsiteSaleDelivery, self).update_eshop_carrier(**post)
```
Comment (nim-odoo) [inline addons/website_sale_delivery/controllers/main.py]: unreliable since `website_sale_coupon` and `website_sale_delivery` do not depend on each other.

**Comment by nim-odoo** [comment]
As @JKE-be said, the diff is huge and not nice. Moreover, the `super` calls are unreliable because of the lack of dependencies between the two modules: you never know which method will be called first.

So either:
- move the controller in website_sale, and call specific method to make sure of the call order
- make a specific module
- use the "hack dégueulasse" which I am not aware of

**Comment by kebeclibre** [comment]
better PR at https://github.com/odoo/odoo/pull/26935

**Pull request closed without merging**